nixos/modules/services/monitoring/prometheus-node-exporter.nix

```diff
@@ -0,0 +1,49 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.prometheus-node-exporter;
+  promUser = "prometheus-node-exporter";
+  cmdlineArgs = cfg.extraFlags ++ [
+    "-web.listen-address=${cfg.listenAddress}"
+  ];
+in {
+  options = {
+    services.prometheus-node-exporter = {
+      enable = mkEnableOption "Enable the Prometheus node exporter (CPU stats etc).";
+      listenAddress = mkOption {
+        type = types.str;
+        default = "0.0.0.0:9100";
+        description = ''
+          Address to listen on.
+        '';
+      };
+      extraFlags = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = ''
+          Extra commandline options when launching the node exporter.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.extraUsers."${promUser}" = {
+      description = "Prometheus node exporter daemon user";
+    };
+    systemd.services.prometheus-node-exporter = {
+      wantedBy = [ "multi-user.target" ];
+      after    = [ "network.target" ];
+      script = ''
+        exec ${pkgs.prometheus-node-exporter}/bin/node_exporter \
+        ${concatStringsSep " \\\n  " cmdlineArgs}
+      '';
+      serviceConfig = {
+        User = promUser;
```

Comment:
I think we don't really need an extra user for the node exporter as it doesn't have to keep any state on the filsystem. I think `nobody` should do just fine.